app/services/sync/fetch_repos.rb

```diff
@@ -7,14 +7,35 @@ def initialize(profile)
 
     def call
       github_repos = Github::ReposConsumer.new(profile.nickname).call
-      github_repos.each do |github_repo|
-        repo = Sync::SyncRepo.new(profile, github_repo).create_repo
-        Sync::FetchLanguages.new(profile, repo).call
-      end
+      github_repos.each { |github_repo| syncronize_repo(github_repo) }
     end
 
     private
 
-    attr_accessor :profile, :repos
+    attr_accessor :profile, :repos, :repo
+
+    def syncronize_repo(github_repo)
+      self.repo = Repo.find_by(name: github_repo[:name])
+
+      create_repo_if_not_exists(github_repo)
+      syncronyze_repo(github_repo)
+    end
+
+    def create_repo_if_not_exists(github_repo)
+      return unless repo.nil?
+
+      self.repo = Sync::SyncRepo.new(profile, github_repo).create_repo
+      Sync::FetchLanguages.new(profile, repo).call
+    end
+
+    def syncronyze_repo(github_repo)
+      this_repo_date = repo.git_date
```

Comment:
feels very weird naming the vars with `this_*`. You already had stopped doing this a long ago on js
```suggestion
      repo_date = repo.git_date
```